iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.cs

```diff
@@ -1,101 +1,111 @@
-            var runtimeInfo = await s_eventHubClient.GetRuntimeInformationAsync();
-            var d2cPartitions = runtimeInfo.PartitionIds;
-
-            CancellationTokenSource cts = new CancellationTokenSource();
-
-            Console.CancelKeyPress += (s, e) =>
-            {
-                e.Cancel = true;
-                cts.Cancel();
-                Console.WriteLine("Exiting...");
-            };
-
-            var tasks = new List<Task>();
-            foreach (string partition in d2cPartitions)
-            {
-                tasks.Add(ReceiveMessagesFromDeviceAsync(partition, cts.Token));
-            }
-
-            // Wait for all the PartitionReceivers to finsih.
-            Task.WaitAll(tasks.ToArray());
-        }
-    }
-}
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// This application uses the Azure Event Hubs Client for .NET
+// For samples see: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/README.md
+// For documentation see: https://docs.microsoft.com/azure/event-hubs/
+
+using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+
+namespace read_d2c_messages
+{
+    public class ReadDeviceToCloudMessages
+    {
+        // Event Hub-compatible endpoint
+        // az iot hub show --query properties.eventHubEndpoints.events.endpoint --name {your IoT Hub name}
+        private const string EventHubsCompatibleEndpoint = "{your Event Hubs compatible endpoint}";
+
+        // Event Hub-compatible name
+        // az iot hub show --query properties.eventHubEndpoints.events.path --name {your IoT Hub name}
+        private const string EventHubName = "{your Event Hubs compatible name}";
+
+        // az iot hub policy show --name service --query primaryKey --hub-name {your IoT Hub name}
+        private const string IotHubSasKeyName = "service";
+        private const string IotHubSasKey = "{your service primary key}";
+
+        // Asynchronously create a PartitionReceiver for a partition and then start
+        // reading any messages sent from the simulated client.
+        private static async Task ReceiveMessagesFromDeviceAsync(CancellationToken cancellationToken)
+        {
+            // If you chose to copy the "Event Hub-compatible endpoint" from the "Built-in endpoints" section
+            // of your IoT Hub instance in the Azure portal, you can set the connection string to that value
+            // directly and remove the call to "BuildEventHubsConnectionString".
+            string connectionString = BuildEventHubsConnectionString(EventHubsCompatibleEndpoint, IotHubSasKeyName, IotHubSasKey);
+
+            // Create the consumer using the default consumer group using a direct connection to the service.

```

Comment:
There was a comment left at one point, mentioning a desire to include a commented section with proxy use details.   While I'd prefer to avoid that, as I believe that it hurts readability and adds complexity to the sample that aren't needed in the majority case,  I did add a call-out in the comment to point interested parties at the discussion and snippet for proxy support.

Reply:
Note that this link will not work until this PR has merged, as it refers to a new README asset being added in the scope of these changes.

Reply:
Hi Jesse -- we've included the code in Node and Java samples, but you did a great job explaining it in the Readme. I'm OK with not having it in the sample for now. @ramya-rao-a - cool?